torch/_inductor/scheduler.py

```diff
@@ -1416,9 +1445,16 @@ def remove_kernel_local_buffers(self):
         same kernel can be removed.
         """
 
-        names_to_remove = (
-            V.kernel.store_buffer_names & self.buffer_names_no_longer_needed
-        )
+        fused_node_ids = {
+            node.node_idx
+            for node in V.kernel.node_schedule
+            if isinstance(node, BaseSchedulerNode)
+        }
+        names_to_remove = []
+        for out_buf in V.kernel.store_buffer_names:
+            users = self.buffer_users[out_buf]
+            if users.issubset(fused_node_ids):
+                names_to_remove.append(out_buf)
```

Comment:
Is this not equivalent to the following, without needing the new `buffer_users` dict?

```python
        fused_nodes = [
            node for node in V.kernel.node_schedule
            if isinstance(node, BaseSchedulerNode)
        ]
        fused_node_names = {node.get_name() for node in fused_nodes}
        names_to_remove = {
            node.get_name()
            for node in fused_nodes
            if {u.get_name() for u in node.users}.issubset(fused_node_names)
        }
```

Reply:
oh, I didn't realize that we already have `node.users` tracking the dependencies between nodes. I'll double check but most likely I can avoid add `buffer_users` dict.